Sources/StreamChatUI/Utils/AudioPlayer/AudioPlayerObserving.swift

```diff
@@ -0,0 +1,132 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import AVFoundation
+import Foundation
+
+/// Describes an object that can provide AVPlayer related updates.
+public protocol AudioPlayerObserving {
+    /// Registers an observer with the player that will receive updates when the player's
+    /// ``timeControlStatus`` gets updated.
+    /// - Parameters:
+    /// player: The player from which we would like to receive updates
+    /// block: The block to call once a ``timeControlStatus`` update occurs
+    func addTimeControlStatusObserver(
+        _ player: AVPlayer,
+        using block: @escaping (AVPlayer.TimeControlStatus?) -> Void
+    )
+
+    /// Registers an observer that will periodically invoke the given block during playback to
+    /// report changing time.
+    /// - Parameters:
+    /// player: The player from which we would like to receive updates
```

Comment:
Doc formatting is not good at several places - we might have trouble generating the DocC docs.

Reply:
The shortcut doesn't work for me. What exactly is it generating?